src/targeting.js

```diff
@@ -49,25 +49,41 @@ export function newTargeting(auctionManager) {
         });
       });
     });
+
+    targeting = targeting.map(targeting => {
+      return {
+        [Object.keys(targeting)[0]]: targeting[Object.keys(targeting)[0]]
+          .map(target => {
+            return {
+              [Object.keys(target)[0]]: target[Object.keys(target)[0]].join(', ')
+            };
+          }).reduce((p, c) => Object.assign(c, p), {})
+      };
+    }).reduce(function (accumulator, targeting) {
+      var key = Object.keys(targeting)[0];
+      accumulator[key] = Object.assign({}, accumulator[key], targeting[key]);
+      return accumulator;
+    }, {});
+
     return targeting;
   };
 
   targeting.setTargeting = function(targetingConfig) {
     window.googletag.pubads().getSlots().forEach(slot => {
-      targetingConfig.filter(targeting => Object.keys(targeting)[0] === slot.getAdUnitPath() ||
-        Object.keys(targeting)[0] === slot.getSlotElementId())
-        .forEach(targeting => targeting[Object.keys(targeting)[0]]
-          .forEach(key => {
-            key[Object.keys(key)[0]]
-              .map((value) => {
-                utils.logMessage(`Attempting to set key value for slot: ${slot.getSlotElementId()} key: ${Object.keys(key)[0]} value: ${value}`);
-                return value;
-              })
-              .forEach(value => {
-                slot.setTargeting(Object.keys(key)[0], value);
-              });
-          }));
-    });
+      Object.keys(targetingConfig).filter(targetId => targetId === slot.getSlotElementId() || targetId === slot.getAdUnitPath())
```

Comment:
combine this check into a function please and replace the other occurrence of it.